resources/util.ts

```diff
@@ -176,6 +183,196 @@ const watchCombatant: WatchCombatantFunc = (params, func) => {
+  | 'dirNNW'
+  | 'unknown';
+
+export type DirectionOutput8 =
+  | 'dirN'
+  | 'dirNE'
+  | 'dirE'
+  | 'dirSE'
+  | 'dirS'
+  | 'dirSW'
+  | 'dirW'
+  | 'dirNW'
+  | 'unknown';
+
+export type DirectionOutputCardinal =
+  | 'dirN'
+  | 'dirE'
+  | 'dirS'
+  | 'dirW'
+  | 'unknown';
+
+export type DirectionOutputIntercard =
+  | 'dirNE'
+  | 'dirSE'
+  | 'dirSW'
+  | 'dirNW'
+  | 'unknown';
+
+export type DirectionOutput16Map = { [key in DirectionOutput16]: OutputStrings };
+export type DirectionOutput8Map = { [key in DirectionOutput8]: OutputStrings };
+export type DirectionOutputCardinalMap = { [key in DirectionOutputCardinal]: OutputStrings };
+export type DirectionOutputIntercardMap = { [key in DirectionOutputIntercard]: OutputStrings };
+
+export const output8Dir: DirectionOutput8[] = [
+  'dirN',
+  'dirNE',
+  'dirE',
+  'dirSE',
+  'dirS',
+  'dirSW',
+  'dirW',
+  'dirNW',
+];
+export const outputCardinalDir: DirectionOutputCardinal[] = ['dirN', 'dirE', 'dirS', 'dirW'];
+export const outputIntercardDir: DirectionOutputIntercard[] = ['dirNE', 'dirSE', 'dirSW', 'dirNW'];
+
+const outputStrings16Dir: DirectionOutput16Map = {
+  dirN: Outputs.dirN,
+  dirNNE: Outputs.dirNNE,
+  dirNE: Outputs.dirNE,
+  dirENE: Outputs.dirENE,
+  dirE: Outputs.dirE,
+  dirESE: Outputs.dirESE,
+  dirSE: Outputs.dirSE,
+  dirSSE: Outputs.dirSSE,
+  dirS: Outputs.dirS,
+  dirSSW: Outputs.dirSSW,
+  dirSW: Outputs.dirSW,
+  dirWSW: Outputs.dirWSW,
+  dirW: Outputs.dirW,
+  dirWNW: Outputs.dirWNW,
+  dirNW: Outputs.dirNW,
+  dirNNW: Outputs.dirNNW,
+  unknown: Outputs.unknown,
+};
+
+const outputStrings8Dir: DirectionOutput8Map = {
+  dirN: Outputs.dirN,
+  dirNE: Outputs.dirNE,
+  dirE: Outputs.dirE,
+  dirSE: Outputs.dirSE,
+  dirS: Outputs.dirS,
+  dirSW: Outputs.dirSW,
+  dirW: Outputs.dirW,
+  dirNW: Outputs.dirNW,
+  unknown: Outputs.unknown,
+};
+
+const outputStringsCardinalDir: DirectionOutputCardinalMap = {
+  dirN: Outputs.dirN,
+  dirE: Outputs.dirE,
+  dirS: Outputs.dirS,
+  dirW: Outputs.dirW,
+  unknown: Outputs.unknown,
+};
+
+const outputStringsIntercardDir: DirectionOutputIntercardMap = {
+  dirNE: Outputs.dirNE,
+  dirSE: Outputs.dirSE,
+  dirSW: Outputs.dirSW,
+  dirNW: Outputs.dirNW,
+  unknown: Outputs.unknown,
+};
+
+// TODO: Accept 'north' as a function input and adjust output accordingly.
+// E.g. Math.round((north + 4) - 4 * Math.atan2(x, y) / Math.PI) % 8;
+// Will need to adjust the output arrays as well though.
+
+const xyTo8DirNum = (x: number, y: number, centerX: number, centerY: number): number => {
+  // N = 0, NE = 1, ..., NW = 7
+  x = x - centerX;
+  y = y - centerY;
+  return Math.round(4 - 4 * Math.atan2(x, y) / Math.PI) % 8;
+};
+
+const hdgTo8DirNum = (heading: number): number => {
+  // N = 0, NE = 1, ..., NW = 7
+  return Math.round(4 - 4 * heading / Math.PI) % 8;
+};
+
+const outputFrom8DirNum = (dirNum: number): DirectionOutput8 => {
+  return output8Dir[dirNum] ?? 'unknown';
+};
+
+export const Directions = {
```

Comment:
One last thing that's not obvious here, but can you import `Directions` into `user_config` so that it's available in user triggers as well? See the import for `Util` on line 30/31 where it gets imported with an underscore (see comment above as to why) and then renamed.

Reply:
Done!